### ISSUE TITLE
fatal error: runtime: out of memory

### ISSUE THREAD

**Issue by Dr-Terrible** [muesli/beehive#175]
When manually installing the git HEAD I got an OOM error after the command `make build` consumed around 5GB of my memory (I have 4GB of RAM only):

```shell
go build -tags 'embed' -ldflags '-s -w -X main.Version=0.2.99 -X main.CommitSHA='
# github.com/muesli/beehive/api
fatal error: runtime: out of memory

runtime stack:
runtime.throw(0xb64733, 0x16)
	/usr/lib/go/src/runtime/panic.go:605 +0x95
runtime.sysMap(0xc4f33d0000, 0x2000000, 0x0, 0xf0a378)
	/usr/lib/go/src/runtime/mem_linux.go:216 +0x1d0
runtime.(*mheap).sysAlloc(0xee8f40, 0x2000000, 0x0)
	/usr/lib/go/src/runtime/malloc.go:470 +0xd7
runtime.(*mheap).grow(0xee8f40, 0x1000, 0x0)
	/usr/lib/go/src/runtime/mheap.go:887 +0x60
runtime.(*mheap).allocSpanLocked(0xee8f40, 0x1000, 0xf0a3a0, 0x0)
	/usr/lib/go/src/runtime/mheap.go:800 +0x334
runtime.(*mheap).allocManual(0xee8f40, 0x1000, 0xf0a3a0, 0x0)
	/usr/lib/go/src/runtime/mheap.go:762 +0x53
runtime.stackalloc(0x2000000, 0x0, 0x0)
	/usr/lib/go/src/runtime/stack.go:384 +0x21f
runtime.copystack(0xc420000180, 0x2000000, 0xc400000000)
	/usr/lib/go/src/runtime/stack.go:844 +0x7d
runtime.shrinkstack(0xc420000180)
	/usr/lib/go/src/runtime/stack.go:1145 +0x100
runtime.scanstack(0xc420000180, 0xc42004a560)
	/usr/lib/go/src/runtime/mgcmark.go:770 +0x1a3
runtime.newstack(0x0)
	/usr/lib/go/src/runtime/stack.go:1025 +0x3a1
runtime.morestack()
	/usr/lib/go/src/runtime/asm_amd64.s:415 +0x86

goroutine 1 [GC assist marking (scan)]:
runtime.rawstring(0x3625c, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/lib/go/src/runtime/string.go:244 +0x9c fp=0xc472723da0 sp=0xc472723d98 pc=0x44334c
runtime.rawstringtmp(0x0, 0x3625c, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/lib/go/src/runtime/string.go:115 +0x6f fp=0xc472723de0 sp=0xc472723da0 pc=0x442d0f
runtime.concatstrings(0x0, 0xc472723ec0, 0x3, 0x3, 0xc472723f40, 0x410376)
	/usr/lib/go/src/runtime/string.go:46 +0x10a fp=0xc472723e78 sp=0xc472723de0 pc=0x44283a
runtime.concatstring3(0x0, 0xc4f32aa000, 0x36240, 0x0, 0x0, 0xc423180a60, 0x1c, 0xc422c7e1e0, 0xc4af402160)
	/usr/lib/go/src/runtime/string.go:59 +0x47 fp=0xc472723eb8 sp=0xc472723e78 pc=0x442aa7
strings.Join(0xc46543ec20, 0x2, 0x2, 0x0, 0x0, 0xc46543ec20, 0x1)
	/usr/lib/go/src/strings/strings.go:473 +0x43d fp=0xc472723f80 sp=0xc472723eb8 pc=0x4b3f0d
cmd/compile/internal/gc.evconst(0xc42318aa80)
	/usr/lib/go/src/cmd/compile/internal/gc/const.go:676 +0x437 fp=0xc472724280 sp=0xc472723f80 pc=0x95d667
cmd/compile/internal/gc.typecheck1(0xc42318aa80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:2215 +0x391 fp=0xc4727250c0 sp=0xc472724280 pc=0xa3f781
cmd/compile/internal/gc.typecheck(0xc42318aa80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472725250 sp=0xc4727250c0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318ab80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472726090 sp=0xc472725250 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318ab80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472726220 sp=0xc472726090 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318ac80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472727060 sp=0xc472726220 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318ac80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc4727271f0 sp=0xc472727060 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318ad80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472728030 sp=0xc4727271f0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318ad80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc4727281c0 sp=0xc472728030 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318ae80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472729000 sp=0xc4727281c0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318ae80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472729190 sp=0xc472729000 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318af80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472729fd0 sp=0xc472729190 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318af80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47272a160 sp=0xc472729fd0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318b080, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47272afa0 sp=0xc47272a160 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318b080, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47272b130 sp=0xc47272afa0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318b180, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47272bf70 sp=0xc47272b130 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318b180, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47272c100 sp=0xc47272bf70 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318b280, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47272cf40 sp=0xc47272c100 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318b280, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47272d0d0 sp=0xc47272cf40 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318b380, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47272df10 sp=0xc47272d0d0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318b380, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47272e0a0 sp=0xc47272df10 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318b480, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47272eee0 sp=0xc47272e0a0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318b480, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47272f070 sp=0xc47272eee0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318b580, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47272feb0 sp=0xc47272f070 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318b580, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472730040 sp=0xc47272feb0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318b680, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472730e80 sp=0xc472730040 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318b680, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472731010 sp=0xc472730e80 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318b780, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472731e50 sp=0xc472731010 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318b780, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472731fe0 sp=0xc472731e50 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318b880, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472732e20 sp=0xc472731fe0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318b880, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472732fb0 sp=0xc472732e20 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318b980, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472733df0 sp=0xc472732fb0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318b980, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472733f80 sp=0xc472733df0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318ba80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472734dc0 sp=0xc472733f80 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318ba80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472734f50 sp=0xc472734dc0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318bb80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472735d90 sp=0xc472734f50 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318bb80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472735f20 sp=0xc472735d90 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318bc80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472736d60 sp=0xc472735f20 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318bc80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472736ef0 sp=0xc472736d60 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318bd80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472737d30 sp=0xc472736ef0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318bd80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472737ec0 sp=0xc472737d30 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318be80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472738d00 sp=0xc472737ec0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318be80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472738e90 sp=0xc472738d00 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc42318bf80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472739cd0 sp=0xc472738e90 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc42318bf80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472739e60 sp=0xc472739cd0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190080, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47273aca0 sp=0xc472739e60 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190080, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47273ae30 sp=0xc47273aca0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190180, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47273bc70 sp=0xc47273ae30 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190180, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47273be00 sp=0xc47273bc70 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190280, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47273cc40 sp=0xc47273be00 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190280, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47273cdd0 sp=0xc47273cc40 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190380, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47273dc10 sp=0xc47273cdd0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190380, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47273dda0 sp=0xc47273dc10 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190480, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47273ebe0 sp=0xc47273dda0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190480, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47273ed70 sp=0xc47273ebe0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190580, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47273fbb0 sp=0xc47273ed70 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190580, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47273fd40 sp=0xc47273fbb0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190680, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472740b80 sp=0xc47273fd40 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190680, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472740d10 sp=0xc472740b80 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190780, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472741b50 sp=0xc472740d10 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190780, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472741ce0 sp=0xc472741b50 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190880, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472742b20 sp=0xc472741ce0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190880, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472742cb0 sp=0xc472742b20 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190980, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472743af0 sp=0xc472742cb0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190980, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472743c80 sp=0xc472743af0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190a80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472744ac0 sp=0xc472743c80 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190a80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472744c50 sp=0xc472744ac0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190b80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472745a90 sp=0xc472744c50 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190b80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472745c20 sp=0xc472745a90 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190c80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472746a60 sp=0xc472745c20 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190c80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472746bf0 sp=0xc472746a60 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190d80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472747a30 sp=0xc472746bf0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190d80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472747bc0 sp=0xc472747a30 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190e80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472748a00 sp=0xc472747bc0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190e80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472748b90 sp=0xc472748a00 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423190f80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc4727499d0 sp=0xc472748b90 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423190f80, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472749b60 sp=0xc4727499d0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423191080, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47274a9a0 sp=0xc472749b60 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423191080, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47274ab30 sp=0xc47274a9a0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423191180, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47274b970 sp=0xc47274ab30 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423191180, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47274bb00 sp=0xc47274b970 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423191280, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47274c940 sp=0xc47274bb00 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423191280, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47274cad0 sp=0xc47274c940 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423191380, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47274d910 sp=0xc47274cad0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423191380, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47274daa0 sp=0xc47274d910 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423191480, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47274e8e0 sp=0xc47274daa0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423191480, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47274ea70 sp=0xc47274e8e0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423191580, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc47274f8b0 sp=0xc47274ea70 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423191580, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc47274fa40 sp=0xc47274f8b0 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423191680, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472750880 sp=0xc47274fa40 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423191680, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc472750a10 sp=0xc472750880 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423191780, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472751850 sp=0xc472750a10 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423191780, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc4727519e0 sp=0xc472751850 pc=0xa3eeb4
cmd/compile/internal/gc.typecheck1(0xc423191880, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:569 +0x33d3 fp=0xc472752820 sp=0xc4727519e0 pc=0xa427c3
cmd/compile/internal/gc.typecheck(0xc423191880, 0x2, 0x0)
	/usr/lib/go/src/cmd/compile/internal/gc/typecheck.go:201 +0x764 fp=0xc4727529b0 sp=0xc472752820 pc=0xa3eeb4
...additional frames elided...
```

How is possible that beehive's source code needs so much memory during compilation? I can easily compile packages that are way way more complex and bigger than beehive, such as JuJu or Docker. And they don't consume so much RAM, even when I'm running race tests or pprofiling.

**Comment by muesli** [comment]
You could disable a bunch of hives in hives.go (simply comment them out with a leading `//`). We'll start providing binaries with the next release, if that's an acceptable solution for you.

**Comment by muesli** [comment]
That being said, I can compile Beehive just fine on a Raspberry Pi (ARM) with 1GB of RAM, as well as on an APU (x86_64) with 2GB.

**Comment by Dr-Terrible** [comment]
Hi @muesli,
I am sorry to have kept you waiting for so long for an answer.
The culprit for the reported issue turned out to be `go-bindata`, for the reasons I explained in this bug report: Dr-Terrible/go-overlay#76

TL;DR: the original `github.com/jteeuwen/go-bindata` package doesn't play very well with recent Go toolchains (v1.10+); some Linux distributions, like Gentoo and Arch, decided to package a fork of `go-bindata` with all the fixes for the new Go toolchain, most notably `github.com/shuLhan/go-bindata` (v3.2.0) which also triggered the memory error I reported here. Subsequent version (v3.3.0) fixed it, so this report can be closed. At a certain point, Jim Teeuwen removed himself from GitHub and all his packages disappeared.

As the official maintainer of Beehive for Gentoo/Linux I suggest you to abandon `go-bindata` in favour of better solutions, such as:

- github.com/shurcooL/vfsgen  (highly recommended)
- github.com/rakyll/statik
- github.com/mjibson/esc
- github.com/gobuffalo/packr

If you really need to stick to `go-bindata`, then there are two official forks that maintain API/ABI compatibility:
- github.com/kevinburke/go-bindata (highly recommended)
- github.com/shuLhan/go-bindata
